### PR TITLE
Add allure reporting, new way of cleaning up db, more tests, new way that objects are compared in tests

### PR DESCRIPTION
**Pull request by hdlugosz**

### CODE REVIEW

**Comment by hdlugosz** [comment]
Here is an example test result using allure
allure2
allure1
allure3

**Comment by hdlugosz** [comment]
In addition, I decided to use the http code 422 in places where the user provides data in the correct format, but for some reason they are rejected by the application - for example, an empty name field or a too short password and code 409 in places where the user tries to add resource that already exists in the database - for example, a user with such a login already exists.

File: src/main/java/vikta/database/DatabaseController.java [hunk 1]
```diff
@@ -158,4 +164,83 @@ private boolean exists(String query, int id) {
...
+
+        for (int id : userIDs) {
+            deleteUserByID(id);
+        }
+        for (int id : addressIDs) {
+            deleteAddressByID(id);
+        }
+        for (int id : paymentCardIDs) {
+            deletePaymentCardByID(id);
+        }
```
Comment (kateryna-skakunova) [inline src/main/java/vikta/database/DatabaseController.java]: any experience working with lambdas in Java? try to rewrite it using forEach

File: src/main/java/vikta/database/DatabaseController.java [hunk 2]
```diff
@@ -158,4 +164,83 @@ private boolean exists(String query, int id) {
...
+    public ArrayList<Integer> listUserIDsToBeDeleted(int latestUserID) {
+            return listIDsToBeDeleted("SELECT id FROM user WHERE id > ?;", latestUserID);
+    }
+
+    public ArrayList<Integer> listAddressIDsToBeDeleted(int latestAddressID) {
+            return listIDsToBeDeleted("SELECT id FROM address WHERE id > ?;", latestAddressID);
+    }
+
+    public ArrayList<Integer> listPaymentCardIDsToBeDeleted(int latestPaymentCardID) {
+            return listIDsToBeDeleted("SELECT id FROM paymentcard WHERE id > ?;", latestPaymentCardID);
```
Comment (kateryna-skakunova) [inline src/main/java/vikta/database/DatabaseController.java:222]: formatting?

**Review by kateryna-skakunova — APPROVED** [review]

**Pull request merged**